orbit/pkg/update/disk_encryption.go

```diff
@@ -0,0 +1,41 @@
+package update
+
+import (
+	"github.com/fleetdm/fleet/v4/orbit/pkg/useraction"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+)
+
+const maxRetries = 2
+
+type DiskEncryptionRunner struct {
+	Fetcher OrbitConfigFetcher
+
+	isIdle bool
+}
+
+func ApplyDiskEncryptionRunnerMiddleware(f OrbitConfigFetcher) DiskEncryptionRunner {
+	return DiskEncryptionRunner{Fetcher: f}
+}
+
+func (d DiskEncryptionRunner) GetConfig() (*fleet.OrbitConfig, error) {
+	cfg, err := d.Fetcher.GetConfig()
+	if err != nil {
+		log.Info().Err(err).Msg("calling GetConfig from DiskEncryptionFetcher")
+		return nil, err
+	}
+
+	log.Debug().Msgf("running disk encryption fetcher middleware, notification: %v, isIdle: %v", cfg.Notifications.RotateDiskEncryptionKey, d.isIdle)
+
+	if cfg.Notifications.RotateDiskEncryptionKey && !d.isIdle {
+		d.isIdle = true
+		go func() {
+			defer func() { d.isIdle = false }()
```

Comment:
This `d.isIdle` field access is racy, it should be protected with a mutex, or since it is just a bool, I think an `atomic.Bool` (https://pkg.go.dev/sync/atomic#Bool) would work fine. However, it seems like the name (or the value) is backwards? Ie. it is set to `true` when it is _not_ idle, or am I misunderstanding the intent?